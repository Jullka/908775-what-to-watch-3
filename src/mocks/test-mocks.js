const promoFilm = {
  title: `Mock-test-The Grand Budapest Hotel`,
  genre: `Mock-test-Drama`,
  releaseDate: 2020,
};

const filmDetails = {
  id: `044`,
  title: `Terminator 2: Judgment Day`,
  genre: `Thrillers`,
  releaseDate: 1991,
  poster: `img/johnny-english.jpg`,
  bigPoster: `img/moonrise-kingdom.jpg`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  ratingScore: 8.9,
  ratingLevel: `Very good`,
  ratingCount: 240,
  runTime: `154`,
  text: `Based on a true story about a small-time, self-possessed personal-injury attorney whose greed entangles him in a case that threatens to destroy him. The Woburn Case- which appears straightforward- instead evolves into a labyrinthine lawsuit of epic proportions where truth, if it can be found at all, resides not in the courtroom, but buried deep in a network of deceit and corruptions.`,
  director: `Steven Zaillian`,
  starring: [`John Travolta`, `Robert Duvall`, `Stephen Fry`],
  reviews: [
    {
      text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
      author: {
        id: 1,
        name: `Kate Muir`
      },
      date: `March 15, 2019`,
      rating: 9.0
    },
    {
      text: `Ut ornare lectus sit amet. Molestie at elementum eu facilisis sed odio morbi quis. Sit amet dictum sit amet justo donec. Pulvinar neque laoreet suspendisse interdum consectetur libero id faucibus nisl.`,
      author: {
        id: 2,
        name: `Bill Goodykoontz`
      },
      date: `March 25, 2019`,
      rating: 8.0
    },
    {
      id: `3`,
      text: `Pulvinar elementum integer enim neque volutpat ac tincidunt. Parturient montes nascetur ridiculus mus mauris vitae.`,
      author: {
        id: 3,
        name: `Amanda Greever`
      },
      date: `March 15, 2018`,
      rating: 6.7
    },
  ]
};

const films = [
  {
    id: `001`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    genre: `Thrillers`,
    releaseDate: 1991,
    poster: `img/johnny-english.jpg`,
    video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `002`,
    title: `Bohemian Rhapsody`,
    genre: `Thrillers`,
    releaseDate: 1991,
    poster: `img/johnny-english.jpg`,
    video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `003`,
    title: `Macbeth`,
    genre: `Thrillers`,
    releaseDate: 1991,
    poster: `img/johnny-english.jpg`,
    video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `004`,
    title: `Aviator`,
    genre: `Thrillers`,
    releaseDate: 1991,
    poster: `img/johnny-english.jpg`,
    video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `005`,
    title: `We need to talk about Kevin`,
    genre: `Thrillers`,
    releaseDate: 1991,
    poster: `img/johnny-english.jpg`,
    video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `006`,
    title: `What We Do in the Shadows`,
    genre: `Thrillers`,
    releaseDate: 1991,
    poster: `img/johnny-english.jpg`,
    video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `007`,
    title: `Revenant`,
    genre: `Thrillers`,
    releaseDate: 1991,
    poster: `img/johnny-english.jpg`,
    video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `008`,
    title: `Johnny English`,
    genre: `Thrillers`,
    releaseDate: 1991,
    poster: `img/johnny-english.jpg`,
    video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  }
];

export {promoFilm, filmDetails, films};

