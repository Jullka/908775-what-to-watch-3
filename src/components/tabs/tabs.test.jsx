import React from 'react';
import renderer from 'react-test-renderer';
import Tabs from '../tabs/tabs.jsx';
import withActiveItem from '../hocs/with-active-item/with-active-item.jsx';
import {filmDetails} from '../../mocks/test-mocks.js';


const TabsWrapped = withActiveItem(Tabs);

it(`Tabs should render correctly`, () => {
  const tree = renderer
    .create(
        <TabsWrapped
          film={filmDetails}
        />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
