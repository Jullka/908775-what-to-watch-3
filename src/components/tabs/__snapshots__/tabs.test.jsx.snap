// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should render correctly 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item "
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item "
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="movie-rating movie-tab movie-tab--overview
        "
  >
    <div
      className="movie-rating__score"
    />
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        
      </span>
      <span
        className="movie-rating__count"
      >
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text movie-tab movie-tab--overview
        "
  >
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Steven Zaillian
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        John Travolta, Robert Duvall, Stephen Fry
         and other
      </strong>
    </p>
  </div>,
  <div
    className="movie-card__text movie-card__row movie-tab movie-tab--details
        visually-hidden"
  >
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Director
        </strong>
        <span
          className="movie-card__details-value"
        >
          Steven Zaillian
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Starring
        </strong>
        <span
          className="movie-card__details-value"
        >
          John Travolta
           
          <br />
          Robert Duvall
           
          <br />
          Stephen Fry
           
          <br />
        </span>
      </p>
    </div>
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Run Time
        </strong>
        <span
          className="movie-card__details-value"
        >
          NaN
          h 
          NaN
          m
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Genre
        </strong>
        <span
          className="movie-card__details-value"
        >
          Thrillers
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Released
        </strong>
        <span
          className="movie-card__details-value"
        >
          1991
        </span>
      </p>
    </div>
  </div>,
  <div
    className="movie-card__reviews movie-card__row movie-tab movie-tab--reviews
        visually-hidden"
  >
    <div
      className="movie-card__reviews-col"
    >
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Kate Muir
            </cite>
            <time
              className="review__date"
              dateTime="March 15, 2019"
            >
              March 15, 2019
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          9
        </div>
      </div>
      <br />
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Ut ornare lectus sit amet. Molestie at elementum eu facilisis sed odio morbi quis. Sit amet dictum sit amet justo donec. Pulvinar neque laoreet suspendisse interdum consectetur libero id faucibus nisl.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Bill Goodykoontz
            </cite>
            <time
              className="review__date"
              dateTime="March 25, 2019"
            >
              March 25, 2019
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          8
        </div>
      </div>
      <br />
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Pulvinar elementum integer enim neque volutpat ac tincidunt. Parturient montes nascetur ridiculus mus mauris vitae.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Amanda Greever
            </cite>
            <time
              className="review__date"
              dateTime="March 15, 2018"
            >
              March 15, 2018
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          6.7
        </div>
      </div>
      <br />
    </div>
  </div>,
]
`;
